pkg/clients/appconfiguration/keyvalue.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package appconfiguration
+
+import (
+	"github.com/google/go-cmp/cmp"
+	"github.com/mitchellh/copystructure"
+	"github.com/pkg/errors"
+	"github.com/stone-payments/appconfig-go-sdk/appconfig/keyvalues"
+
+	"github.com/crossplane/provider-azure/apis/appconfiguration/v1alpha1"
+)
+
+const (
+	errCheckUpToDate = "unable to determine if external resource is up to date"
+)
+
+// LateInitialize fills the cr values that user did not fill with their
+// corresponding value in the Azure, if there is any.
+func LateInitialize(cr *v1alpha1.KeyValueParameters, p keyvalues.KeyValue) {
+	if cr.ContentType == nil && p.ContentType != nil {
+		cr.ContentType = p.ContentType
+	}
+
+	if cr.Locked == nil && p.Locked != nil {
```

Comment:
You can to use the [late initialize helpers functions](https://github.com/stone-payments/provider-azure/blob/master/pkg/clients/azure.go#L399)